src/query/expression/src/types/number.rs

```diff
@@ -877,6 +877,55 @@ macro_rules! with_number_mapped_type {
     }
 }
 
+#[macro_export]
+macro_rules! with_number_data_types {
```

Comment:
why not use `ALL_NUMERICS_TYPES` with `with_number_mapped_type`

Reply:
This macro deals with two types simultaneously. This macro is from the old codes.

If use `with_number_mapped_type`, we need to write two code blocks.